src/main/kotlin/org/seekerwing/aws/sqsconsumer/messageprovider/SingleQueueBasedMessageProvider.kt

```diff
@@ -0,0 +1,18 @@
+package org.seekerwing.aws.sqsconsumer.messageprovider
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.produce
+import org.seekerwing.aws.sqsconsumer.configuration.MessageProviderConfiguration
+import org.seekerwing.aws.sqsconsumer.model.MessageEnvelope
+import org.seekerwing.aws.sqsconsumer.sqs.launchMessageFetcher
+
+internal class SingleQueueBasedMessageProvider(
+    private val configuration: MessageProviderConfiguration
+) : MessageProvider {
+
+    override fun provideMessage(coroutineScope: CoroutineScope) = coroutineScope.produce<MessageEnvelope> {
+        repeat(configuration.parallelism) {
+            launchMessageFetcher(configuration, channel)
```

Comment:
I am unclear where the channel variable comes from

Reply:
the 

```
repeat(configuration.parallelism) {
  launchMessageFetcher(configuration, channel)
}
```

block is an implementation of the [ProducerScope](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-producer-scope/) interface required by [produce](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/produce.html); the interface makes available the [channel](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-producer-scope/channel.html) variable.